lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1448,30 +1449,31 @@ std::vector<std::pair<Decoration, std::string>>
 tryParseIntelFPGAAnnotationString(StringRef AnnotatedCode) {
   std::vector<std::pair<Decoration, std::string>> Decorates;
 
-  size_t OpenBracketNum = AnnotatedCode.count('{');
```

Comment:
Are you sure that there cannot be other strings with different format except annotation strings with curly brackets?

Reply:
I mean, I can't envision such a situation that could make it into `tryParseIntelFPGAAnnotationString(...)`. Am I missing some cases with var/ptr annotation strings?

Reply:
This function is designed to handle only strings with curly brackets. Incorrect (without curly brackets) string will be handled as UserSemantic decoration

Reply:
Ok, anyway Alexey already approved these changes.

Reply:
I'm ready to merge this PR, but there is a conflict. Please resolve it.

Reply:
@AlexeySotkin, done. I've also updated the patch to match curly-braced sequences (with a subsequent drop of these symbols). Please check if it's still good to go.